tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/RecordingZ3SolverContext.scala

```diff
@@ -0,0 +1,207 @@
+package at.forsyte.apalache.tla.bmcmt.smt
+
+import at.forsyte.apalache.tla.bmcmt.profiler.SmtListener
+import at.forsyte.apalache.tla.bmcmt.{Arena, ArenaCell}
+import at.forsyte.apalache.tla.lir.TlaEx
+import com.typesafe.scalalogging.LazyLogging
+
+
+object RecordingZ3SolverContext {
+  def apply(parentLog: Option[SmtLog], config: SolverConfig): RecordingZ3SolverContext = {
+    val context = new RecordingZ3SolverContext(parentLog, config)
+    parentLog match {
+      case Some(log) =>
+        log.replay(context.solver)
+
+      case None => ()
+    }
```

Comment:
Iiuc, this is [forEach](https://www.scala-lang.org/api/current/scala/Option.html#foreach[U](f:A=%3EU):Unit):

```suggestion
    parentLog.forEach(_.replay(context.solver))
```

Reply:
Very true :D